ci-operator/step-registry/ipi/conf/capability/baselinecaps/vset/ipi-conf-capability-baselinecaps-vset-commands.sh

```diff
@@ -84,12 +99,27 @@ capabilities:
 EOF
 
 #To enable required capablities whatever baselineCapabilitySet setting
+additional_caps=""
 if [[ "${ADDITIONAL_ENABLED_CAPABILITIES}" != "" ]]; then
     echo "Enable required capabilities: ${ADDITIONAL_ENABLED_CAPABILITIES}"
+    additional_caps="${ADDITIONAL_ENABLED_CAPABILITIES}"
+fi
+
+# Capablities dependency
+if [[ "${selected_cap_set}" != "vCurrent" ]]; then
+    # cap marketplace must be enabled along with OperatorLifecycleManager
+    selected_set=${selected_cap_set//.}
+    if [[ "${!selected_set}" =~ "marketplace" ]] && [[ ! "${!selected_set}" =~ "OperatorLifecycleManager" ]] && [[ "${v_current}" =~ "OperatorLifecycleManager" ]]; then
```

Comment:
How about introducing a new array var to define the dependency, so that in the future, we can easily extend it.
Something like:
```
declare -A dependency_caps
dependency_caps["marketplace"]="OperatorLifecycleManager"
for key in in "${!dependency_caps[@]}"; do
    if [[ "${!selected_set}" =~ "$key" ]] && [[ ! "${!selected_set}" =~ " ${dependency_caps[$key]}" ]] && [[ "${v_current}" =~ "${dependency_caps[$key]}" ]]; then
        echo "capability '$key' in capset '${selected_cap_set}' requires ${dependency_caps[$key]}, enabling it together"
        additional_caps="${additional_caps} ${dependency_caps[$key]}"
    fi
done
```

Reply:
good suggestion, updated.